impeller/entity/contents/filters/blend_filter_contents.cc

```diff
@@ -373,6 +374,20 @@ std::optional<Entity> BlendFilterContents::RenderFilter(
   }
 
   if (blend_mode_ <= Entity::kLastAdvancedBlendMode) {
+    auto potential_alpha = GetAlpha().value_or(1.0);
+    if (inputs.size() == 1 && foreground_color_.has_value() &&
+        potential_alpha >= 1.0 - kEhCloseEnough) {
+      auto contents = std::make_shared<AdvancedForegroundBlendContents>();
+      contents->SetBlendMode(blend_mode_);
+      contents->SetCoverage(coverage);
+      contents->SetSrcInput(inputs[0]);
+      contents->SetForegroundColor(foreground_color_.value());
+      Entity entity;
+      entity.SetTransformation(Matrix::MakeTranslation(coverage.origin));
+      entity.SetContents(std::move(contents));
+      return entity;
```

Comment:
To prevent the need for adding new contents that don't really have a use outside of filters (and for just keeping all the logic local), you could also use an `AnonymousContents` here, which I added to help make returning specialized contents easier when optimizing filters like this.
It just takes a render method and coverage, no questions asked. :)

Up to you if you want to do that refactor though.

Reply:
I do intend to use this a bit more though, for some of the follow ups in https://github.com/flutter/flutter/issues/124025 the bypass the color filter entirely.